Prog/NCHousing_needs_units_targets.sas

```diff
@@ -0,0 +1,927 @@
+	set of targets for the future. 
+
+ Modifications: 02-12-19 LH Adjust weights using Calibration from Steven's projections 
+						 	so that occupied units match COG 2015 HH estimation.
+                02-17-19 LH Readjust weights after changes to calibration to move 2 HH w/ GQ=5 out of head of HH
+				03-30-19 LH Remove hard coding and merge in contract rent to gross rent ratio for vacant units. 
+				04-26-19 LH Change halfway from 30% of income to max_rent or max_ocost.
+                01/2020  YS update for NC housing project
+**************************************************************************/
+
+%include "L:\SAS\Inc\StdLocal.sas";
+
+** Define libraries **;
+%DCData_lib( NCHsg )
+%DCData_lib( Ipums )
+
+%let date=12312019; 
+
+proc format;
+
+  value hud_inc
+   .n = 'Vacant'
+    1 = '0-30% AMI'
+    2 = '31-50%'
+    3 = '51-80%'
+    4 = '81-120%'
+    5 = '120-200%'
+    6 = 'More than 200%'
+	;
+
+  value tenure
+    1 = 'Renter units'
+    2 = 'Owner units'
+	;
+/*
+  value county2_char
+    1= "DC"
+	2= "Charles County"
+	3= "Frederick County "
+	4="Montgomery County"
+	5="Prince Georges "
+	6="Arlington"
+	7="Fairfax, Fairfax city and Falls Church"
+	8="Loudoun"
+	9="Prince William, Manassas and Manassas Park"
+    10="Alexandria"
+  	;
+*/
+  value rcost
+	  1= "$0 to $749"
+	  2= "$750 to $1,199"
+	  3= "$1,200 to $1,499"
+	  4= "$1,500 to $1,999"
+	  5= "$2,000 to $2,499"
+	  6= "More than $2,500"
+  ;
+
+  value ocost
+	  1= "$0 to $1,199"
+	  2= "$1,200 to $1,799"
+	  3= "$1,800 to $2,499"
+	  4= "$2,500 to $3,199"
+	  5= "$3,200 to $4,199"
+	  6= "More than $4,200"
+  ;
+
+  value acost
+	  1= "$0 to $799"
+	  2= "$800 to $1,299"
+	  3= "$1,300 to $1,799"
+	  4= "$1,800 to $2,499"
+	  5= "$2,500 to $3,499"
+	  6= "More than $3,500"
+  ;
+	
+  value inc_cat
+
+    1 = '20 percentile'
+    2 = '40 percentile'
+    3 = '60 percentile'
+	4 = '80 percentile'
+	5= '100 percentile'
+    6= 'vacant'
+	;
+  value structure
+	1= 'Single family attached and detached'
+	2= '2-9 units in strucutre'
+	3= '10+ units in strucutre'
+	4= 'Mobile or other'
+	5= 'NA'
+	;
+  	
+  value afford
+
+  1= 'natural affordable (rent < $750)'
+  0= 'not natural affordable'; 
+run;
+
+data crosswalk;
+	set NCHsg.PUMA_county_crosswalk ;
+	county_char= put(county14, 5.);
+	length puma_new $5;
+	puma_new = put(input(cats(puma12),8.),z5.);
+	upuma= "37"||puma_new;
+run;
+
+proc sort data= crosswalk;
+	by upuma;
+run;
+
+data NC17_limits;
+	set NCHsg.NC17_limits;
+run;
+
+proc sort data= NC17_limits;
+	by MID;
+run;
+
+%macro single_year(year);
+
+	data NCvacant_&year._1 ;
+		set Ipums.Acs_&year._vacant_NC ;
+	run;
+
+	proc sort data= NCvacant_&year._1;
+	by upuma;
+	run;
+
+	/*merge IPUMS with crosswalk to county, check #observations*/
+	data NCvacant_&year. ;
+	merge NCvacant_&year._1(in=a) crosswalk;
+	if a;
+	by upuma ;
+	%assign_NCcounty2;
+	%assign_NCcounty3;
+	county2_char = county2;
+
+	run;
+
+
+	data NCarea_&year._1 ;
```

Comment:
same comment as above- this data step and proc sort can be combined in to one proc sort with out=.